contrib/utilities/check_encoding.py

```diff
@@ -20,17 +20,37 @@
 # This script should be invoked from the root folder of the deal.ii
 # repository:
 # contrib/utilities/check_encoding.py
+from __future__ import print_function
 
-import glob
 import itertools
+import io
+import os
+import sys
 
-filenames = itertools.chain(glob.iglob('**/*.h', recursive=True),
-                            glob.iglob('**/*.cc', recursive=True),
-                            glob.iglob('**/*.html', recursive=True))
 
+def filename_generator(suffix):
+    for root, _, file_names in os.walk("./"):
+        for file_name in file_names:
+            if file_name.endswith(suffix):
+                if root == "./":
+                    yield root + file_name
+                else:
+                    yield root + "/" + file_name
+
+
+filenames = itertools.chain(filename_generator(".h"),
+                            filename_generator(".cc"),
+                            filename_generator(".html"))
+
+return_code = 0
 for filename in filenames:
+    file_handle = io.open(filename, encoding='utf-8')
     try:
-        with open(filename, encoding='utf-8') as file:
-            file.read()
-    except:
-        raise Exception(filename + ' is not encoded is not encoded with UTF-8')
+        file_handle.read()
+    except UnicodeDecodeError:
+        print(filename + ' is not encoded with UTF-8')
+        return_code = 1
```

Comment:
I had a lot of trouble getting one error message here instead of multiple messages from swallowed exceptions, so I changed this to simply `print`.

Reply:
I agree, this is a better approach. In order to test the script, I saved two files with `latin1` encoding. The script gives the name of both files. I tested the script with python2.7 and python3.5 and both python versions give the expected result.